apps/dav/lib/Files/Sharing/FilesDropPlugin.php

```diff
@@ -36,57 +36,135 @@ public function enable(): void {
 
 	/**
 	 * This initializes the plugin.
-	 *
-	 * @param \Sabre\DAV\Server $server Sabre server
-	 *
-	 * @return void
-	 * @throws MethodNotAllowed
+	 * It is ONLY initialized by the server on a file drop request.
 	 */
 	public function initialize(\Sabre\DAV\Server $server): void {
 		$server->on('beforeMethod:*', [$this, 'beforeMethod'], 999);
+		$server->on('method:MKCOL', [$this, 'onMkcol']);
 		$this->enabled = false;
 	}
 
-	public function beforeMethod(RequestInterface $request, ResponseInterface $response): void {
+	public function onMkcol(RequestInterface $request, ResponseInterface $response) {
 		if (!$this->enabled || $this->share === null || $this->view === null) {
 			return;
 		}
 
-		// Only allow file drop
+		// If this is a folder creation request we need
+		// to fake a success so we can pretend every
+		// folder now exists.
+		$response->setStatus(201);
+		return false;
+	}
+
+	public function beforeMethod(RequestInterface $request, ResponseInterface $response) {
+		if (!$this->enabled || $this->share === null || $this->view === null) {
+			return;
+		}
+
+		// Retrieve the nickname from the request
+		$nickName = $request->hasHeader('X-NC-Nickname')
```

Comment:
If we could have consistent casing it would be really nice (the word is "nickname not "nick name") :see_no_evil: 
```suggestion
		$nickname = $request->hasHeader('X-NC-Nickname')
```